lib/ts/recipe/thirdpartyemailpassword/recipe.ts

```diff
@@ -0,0 +1,441 @@
+
+    isEmailVerified = async (userId: string) => {
+        return this.emailVerificationRecipe.isEmailVerified(userId, await this.getEmailForUserId(userId));
+    };
+
+    getUsersOldestFirst = async (limit?: number, nextPaginationToken?: string) => {
+        limit = limit === undefined ? 100 : limit;
+        let nextPaginationTokens: {
+            thirdPartyPaginationToken: string | undefined;
+            emailPasswordPaginationToken: string | undefined;
+        } = {
+            thirdPartyPaginationToken: undefined,
+            emailPasswordPaginationToken: undefined,
+        };
+        if (nextPaginationToken !== undefined) {
+            nextPaginationTokens = extractPaginationTokens(this, nextPaginationToken);
+        }
+        let emailPasswordResultPromise = this.emailPasswordRecipe.getUsersOldestFirst(
+            limit,
+            nextPaginationTokens.emailPasswordPaginationToken
+        );
+        let thirdPartyResultPromise =
+            this.thirdPartyRecipe === undefined
+                ? {
+                      users: [],
+                  }
+                : this.thirdPartyRecipe.getUsersOldestFirst(limit, nextPaginationTokens.thirdPartyPaginationToken);
+        let emailPasswordResult = await emailPasswordResultPromise;
+        let thirdPartyResult = await thirdPartyResultPromise;
+        return combinePaginationResults(thirdPartyResult, emailPasswordResult, limit, true);
+    };
+
+    getUsersNewestFirst = async (limit?: number, nextPaginationToken?: string) => {
+        limit = limit === undefined ? 100 : limit;
+        let nextPaginationTokens: {
+            thirdPartyPaginationToken: string | undefined;
+            emailPasswordPaginationToken: string | undefined;
+        } = {
+            thirdPartyPaginationToken: undefined,
+            emailPasswordPaginationToken: undefined,
+        };
+        if (nextPaginationToken !== undefined) {
+            nextPaginationTokens = extractPaginationTokens(this, nextPaginationToken);
+        }
+        let emailPasswordResultPromise = this.emailPasswordRecipe.getUsersNewestFirst(
+            limit,
+            nextPaginationTokens.emailPasswordPaginationToken
+        );
+        let thirdPartyResultPromise =
+            this.thirdPartyRecipe === undefined
+                ? {
+                      users: [],
+                  }
+                : this.thirdPartyRecipe.getUsersNewestFirst(limit, nextPaginationTokens.thirdPartyPaginationToken);
+        let emailPasswordResult = await emailPasswordResultPromise;
+        let thirdPartyResult = await thirdPartyResultPromise;
+        return combinePaginationResults(thirdPartyResult, emailPasswordResult, limit, false);
+    };
+
+    getUserCount = async () => {
+        let promise1 = this.emailPasswordRecipe.getUserCount();
+        let promise2 = this.thirdPartyRecipe !== undefined ? this.thirdPartyRecipe.getUserCount() : 0;
+        return (await promise1) + (await promise2);
+    };
+}
```

Comment:
The sub recipes must not use this recipe's rid when querying the core.. So there needs to be a change for that.

Reply:
done